### PR TITLE
PICARD-2524: prevent crashes when doing multiple searches while one i…

### PR DESCRIPTION
**Pull request by zas**
…s running

# Summary

* This is a…
  * [x] Bug fix
  * [ ] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-2524

# Solution

# Action

### CODE REVIEW

File: picard/ui/searchdialog/album.py [hunk 1]
```diff
@@ -96,6 +99,12 @@ def showEvent(self, event):
         super().showEvent(event)
         self.shown.emit()
 
+    def invalidate(self):
+        self.layout = None
+
+    def __del__(self):
+        self.shown.disconnect()
```
Comment (phw) [inline picard/ui/searchdialog/album.py:106]: Why is this needed? This should happen automatically. Or was there some issue still producible without this change?.

File: picard/ui/searchdialog/album.py [hunk 2]
```diff
@@ -62,6 +62,7 @@ class CoverWidget(QtWidgets.QWidget):
     def __init__(self, parent, width=100, height=100):
         super().__init__(parent)
         self.layout = QtWidgets.QVBoxLayout(self)
+        self.layout.destroyed.connect(self.invalidate)
```
Comment (phw) [inline picard/ui/searchdialog/album.py:65]: I think this could also connect to `self.destroyed`, as it is the widget itself that primarily is the one getting removed. Would have the same effect, though.

**Review by phw — COMMENTED** [review]
Yes, I think this is generally the right approach. It's also what we do in Option pages if there are async tasks modifying UI that might no longer exists at the time the callback gets executed.

File: picard/ui/searchdialog/album.py (hunk 1 above)
Comment (zas) [reply picard/ui/searchdialog/album.py]: It was a leftover from experiments, unneeded.

File: picard/ui/searchdialog/album.py (hunk 2 above)
Comment (zas) [reply picard/ui/searchdialog/album.py]: Yup, done.

**Review by phw — APPROVED** [review]
